presentation/src/main/java/me/androidbox/presentation/login/screen/LoginScreen.kt

```diff
@@ -24,15 +27,38 @@ import me.androidbox.component.general.TaskButton
 import me.androidbox.component.general.UserInputTextField
 import me.androidbox.component.ui.theme.BusbyTaskyTheme
 import me.androidbox.component.ui.theme.backgroundInputEntry
+import me.androidbox.domain.authentication.ResponseState
 import me.androidbox.presentation.login.viewmodel.LoginViewModel
 
 @Composable
 fun LoginScreen(
     modifier: Modifier = Modifier,
     loginViewModel: LoginViewModel = hiltViewModel(),
-    onSignUpClicked: () -> Unit
+    onSignUpClicked: () -> Unit,
+    onLoginSuccess: () -> Unit
 ) {
 
+    val loginState = loginViewModel.loginState.collectAsState()
+
+    LaunchedEffect(key1 = loginState.value) {
+        when(val status = loginState.value) {
+            is ResponseState.Loading -> {
+                /* TODO Showing loading spinner */
+            }
+            is ResponseState.Success -> {
+                loginViewModel.saveCurrentUserDetails(status.data)
+                onLoginSuccess()
+            }
+            is ResponseState.Failure -> {
+                /* Failure */
+                Log.d("LOGIN","${status.error}")
+            }
+            else -> {
+                /* no-op */
+            }
```

Comment:
I personally like to write `else -> Unit` in such case which makes it clear this is a no-op